truss/build.py

```diff
@@ -62,6 +69,73 @@ def mk_truss(
     return scaf
 
 
+def mk_truss_pipeline(
+    pipeline: Callable,
+    target_directory: str = None,
+    bundled_packages: List[str] = None,
+):
+    """Create a Truss from a function. A Truss is a build context designed to
+    be built as a container locally or uploaded into a model serving environment.
+
+    Args:
+        pipeline (a callable function): A function that is expected to be called
+        when the Truss server /predict is invoked.
+        target_directory (str, optional): The local directory target for the Truss. Otherwise a temporary directory
+            will be generated
+        bundled_packages (List[str], optional): Additional local packages that are required by the funtion.
+    Returns:
+        TrussHandle: A handle to the generated Truss that provides easy access to content inside.
+    """
+    if target_directory is None:
+        target_directory_path = build_truss_target_directory("pipeline")
+    else:
+        target_directory_path = Path(target_directory)
+
+    requirements = list(infer_deps(must_include_deps=set(["cloudpickle"])))
+
+    # Create Truss config
+    config = TrussConfig(
+        model_type="custom",
+        model_framework=ModelFrameworkType.CUSTOM,
+        python_version=infer_python_version(),
+        requirements=requirements,
+    )
+
+    # Create data dir
+    (target_directory_path / config.data_dir).mkdir()
+
+    # Create bundled packages dir
+    (target_directory_path / config.bundled_packages_dir).mkdir()
+
+    # Create model module dir via pipeline template
+    model_dir = target_directory_path / config.model_module_dir
+    template_path = TEMPLATES_DIR / "pipeline"
+    copy_tree_path(template_path / "model", model_dir)
+
+    if get_gpu_memory() > 10:
+        click.echo(
```

Comment:
I know it's there at other places as well, but click is a cli tool and use of it here seems weird. This function can be used outside of cli as well. It's fine for now, but please leave a todo here to remove use of click. We could let the function take a log stream as input and write it there, but it's a bit involved, so it's ok to skip that in this PR.